data race in pkg/kubelet

Hit this while [running an unrelated PR through Shippable](https://app.shippable.com/builds/54f0c99f0af4f30e001e846c):

```
W0227 19:54:55.644490   30657 kubelet.go:242] Data dir for pod "bothpod" exists in both old and new form, using new
W0227 19:54:55.646864   30657 kubelet.go:282] Data dir for pod "newpod", container "bothctr" exists in both old and new form, using new
W0227 19:54:55.648438   30657 kubelet.go:977] No ref for pod '1234' - '/k8s_foo_qux_1234_42'
W0227 19:54:55.661498   30657 kubelet.go:977] No ref for pod '1234' - '/k8s_foo_qux_1234_42'
W0227 19:54:55.944516   30657 kubelet.go:977] No ref for pod '1234' - '/k8s_bar_foo.new.test_12345678_0'
W0227 19:54:55.983713   30657 kubelet.go:977] No ref for pod '1234' - '/k8s_foo_bar.new.test_12345678_42'
W0227 19:54:55.992491   30657 kubelet.go:977] No ref for pod '9876' - '/k8s_POD_foo.new.test_12345678_42'
W0227 19:54:56.012539   30657 kubelet.go:977] No ref for pod '7492' - '/k8s_boo_bar.default.testSource_12345678_42'
W0227 19:54:56.012860   30657 kubelet.go:977] No ref for pod '3542' - '/k8s_POD_boo.default.testSource_12345678_42'
W0227 19:54:56.015438   30657 kubelet.go:977] No ref for pod '1234' - '/k8s_foo_bar.new.test_12345678_42'
W0227 19:54:56.015783   30657 kubelet.go:977] No ref for pod '9876' - '/k8s_POD_foo.new.test_12345678_42'
I0227 19:54:56.064712   30657 kubelet.go:2001] Query docker container info for pod bar.new.test failed with error (no containers exist for this pod)
W0227 19:54:56.066136   30657 kubelet.go:977] No ref for pod '1234' - '/k8s_foo_bar.new.test_12345678_1111'
I0227 19:54:56.074874   30657 kubelet.go:2001] Query docker container info for pod foo.new.test failed with error (no containers exist for this pod)
I0227 19:54:56.082731   30657 kubelet.go:1271] pod "foo.new.test" container "bar" hash changed (4660 vs 2854993760). Container will be killed and re-created.
W0227 19:54:56.085344   30657 kubelet.go:977] No ref for pod '1234' - '/k8s_bar.1234_foo.new.test_12345678_42'
W0227 19:54:56.085856   30657 kubelet.go:977] No ref for pod '9876' - '/k8s_POD_foo.new.test_12345678_42'
I0227 19:54:56.105305   30657 kubelet.go:2001] Query docker container info for pod foo.new.test failed with error (no containers exist for this pod)
W0227 19:54:56.113660   30657 probe.go:90] Failed to find probe builder for bar &{Handler:{Exec:<nil> HTTPGet:<nil> TCPSocket:<nil>} InitialDelaySeconds:0 TimeoutSeconds:0}
W0227 19:54:56.124398   30657 probe.go:90] Failed to find probe builder for bar &{Handler:{Exec:<nil> HTTPGet:<nil> TCPSocket:<nil>} InitialDelaySeconds:0 TimeoutSeconds:0}
W0227 19:54:56.124681   30657 probe.go:90] Failed to find probe builder for bar &{Handler:{Exec:<nil> HTTPGet:<nil> TCPSocket:<nil>} InitialDelaySeconds:0 TimeoutSeconds:0}
W0227 19:54:56.128682   30657 kubelet.go:1262] No ref for pod '1234' - 'bar'
I0227 19:54:56.128834   30657 kubelet.go:1268] pod "foo.new.test" container "bar" is unhealthy (probe result: unknown). Container will be killed and re-created.
W0227 19:54:56.129259   30657 kubelet.go:977] No ref for pod '1234' - '/k8s_bar_foo.new.test_12345678_42'
W0227 19:54:56.178639   30657 kubelet.go:512] Unknown protocol "": defaulting to TCP
W0227 19:54:56.179027   30657 kubelet.go:512] Unknown protocol "foobar": defaulting to TCP
W0227 19:54:56.192875   30657 kubelet.go:1637] Pod "..": HostPort is already allocated, ignoring: [[0].port: duplicate value '81']
E0227 19:54:56.281495   30657 kubelet.go:581] Invalid handler: &{<nil> <nil> <nil>}
I0227 19:54:56.284444   30657 kubelet.go:2001] Query docker container info for pod foo.new.test failed with error (no containers exist for this pod)
E0227 19:54:56.294650   30657 kubelet.go:1370] Error running pod "foo.new.test" container "bar": failed to call event handler: test error
E0227 19:54:56.415930   30657 kubelet.go:1214] Unable to get pod with name "foo.new.test" and uid "12345678" info with error(Couldn't find spec for pod foo.new.test)
W0227 19:54:56.662334   30657 probe.go:90] Failed to find probe builder for  <nil>
W0227 19:54:56.676683   30657 probe.go:90] Failed to find probe builder for  <nil>
W0227 19:54:56.678938   30657 probe.go:90] Failed to find probe builder for  <nil>
I0227 19:54:56.707757   30657 runonce.go:62] waiting for 1 pods
I0227 19:54:56.708341   30657 runonce.go:138] container "bar" not found
I0227 19:54:56.708451   30657 runonce.go:112] pod "foo" containers not running: syncing
E0227 19:54:56.713455   30657 kubelet.go:1214] Unable to get pod with name "foo.new.test" and uid "12345678" info with error(Couldn't find spec for pod foo.new.test)
I0227 19:54:56.715598   30657 runonce.go:124] pod "foo" containers synced, waiting for 1s
I0227 19:54:57.719103   30657 runonce.go:138] container "bar" not found
I0227 19:54:57.719221   30657 runonce.go:112] pod "foo" containers not running: syncing
E0227 19:54:57.723001   30657 kubelet.go:1214] Unable to get pod with name "foo.new.test" and uid "12345678" info with error(Couldn't find spec for pod foo.new.test)
I0227 19:54:57.727287   30657 runonce.go:124] pod "foo" containers synced, waiting for 2s
I0227 19:54:59.728595   30657 runonce.go:108] pod "foo" containers running
I0227 19:54:59.728728   30657 runonce.go:75] started pod "foo"
I0227 19:54:59.728778   30657 runonce.go:84] 1 pods started
==================
WARNING: DATA RACE
Read by goroutine 149:
  runtime.mapiterinit()
      /home/shippable/.gvm/gos/go1.4/src/runtime/hashmap.go:535 +0x0
  github.com/docker/spdystream.(*idleAwareFramer).monitor()
      /home/shippable/workspace/src/github.com/GoogleCloudPlatform/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:75 +0x30c

Previous write by goroutine 150:
  runtime.mapassign1()
      /home/shippable/.gvm/gos/go1.4/src/runtime/hashmap.go:376 +0x0
  github.com/docker/spdystream.(*Connection).addStream()
      /home/shippable/workspace/src/github.com/GoogleCloudPlatform/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:837 +0xf6
  github.com/docker/spdystream.(*Connection).addStreamFrame()
      /home/shippable/workspace/src/github.com/GoogleCloudPlatform/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:369 +0x52a
  github.com/docker/spdystream.(*Connection).Serve()
      /home/shippable/workspace/src/github.com/GoogleCloudPlatform/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:261 +0xd06

Goroutine 149 (running) created at:
  github.com/docker/spdystream.NewConnection()
      /home/shippable/workspace/src/github.com/GoogleCloudPlatform/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:184 +0x757
  github.com/GoogleCloudPlatform/kubernetes/pkg/util/httpstream/spdy.NewServerConnection()
      /home/shippable/workspace/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/util/httpstream/spdy/connection.go:54 +0x6e
  github.com/GoogleCloudPlatform/kubernetes/pkg/util/httpstream/spdy.responseUpgrader.UpgradeResponse()
      /home/shippable/workspace/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/util/httpstream/spdy/upgrade.go:71 +0x530
  github.com/GoogleCloudPlatform/kubernetes/pkg/util/httpstream/spdy.(*responseUpgrader).UpgradeResponse()
      <autogenerated>:1 +0xf5
  github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet.(*Server).handleExec()
      github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/_test/_obj_test/server.go:498 +0x943
  github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet.*Server.(github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet.handleExec)·fm()
      github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/_test/_obj_test/server.go:95 +0x52
  net/http.HandlerFunc.ServeHTTP()
      /home/shippable/.gvm/gos/go1.4/src/net/http/server.go:1265 +0x4e
  net/http.(*ServeMux).ServeHTTP()
      /home/shippable/.gvm/gos/go1.4/src/net/http/server.go:1541 +0x20c
  github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet.(*Server).ServeHTTP()
      github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/_test/_obj_test/server.go:733 +0x1f0
  net/http/httptest.(*waitGroupHandler).ServeHTTP()
      /home/shippable/.gvm/gos/go1.4/src/net/http/httptest/server.go:200 +0xf7
  net/http.serverHandler.ServeHTTP()
      /home/shippable/.gvm/gos/go1.4/src/net/http/server.go:1703 +0x1f6
  net/http.(*conn).serve()
      /home/shippable/.gvm/gos/go1.4/src/net/http/server.go:1204 +0x1087

Goroutine 150 (running) created at:
  github.com/GoogleCloudPlatform/kubernetes/pkg/util/httpstream/spdy.newConnection()
      /home/shippable/workspace/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/util/httpstream/spdy/connection.go:68 +0x17a
  github.com/GoogleCloudPlatform/kubernetes/pkg/util/httpstream/spdy.NewServerConnection()
      /home/shippable/workspace/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/util/httpstream/spdy/connection.go:60 +0x117
  github.com/GoogleCloudPlatform/kubernetes/pkg/util/httpstream/spdy.responseUpgrader.UpgradeResponse()
      /home/shippable/workspace/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/util/httpstream/spdy/upgrade.go:71 +0x530
  github.com/GoogleCloudPlatform/kubernetes/pkg/util/httpstream/spdy.(*responseUpgrader).UpgradeResponse()
      <autogenerated>:1 +0xf5
  github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet.(*Server).handleExec()
      github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/_test/_obj_test/server.go:498 +0x943
  github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet.*Server.(github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet.handleExec)·fm()
      github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/_test/_obj_test/server.go:95 +0x52
  net/http.HandlerFunc.ServeHTTP()
      /home/shippable/.gvm/gos/go1.4/src/net/http/server.go:1265 +0x4e
  net/http.(*ServeMux).ServeHTTP()
      /home/shippable/.gvm/gos/go1.4/src/net/http/server.go:1541 +0x20c
  github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet.(*Server).ServeHTTP()
      github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/_test/_obj_test/server.go:733 +0x1f0
  net/http/httptest.(*waitGroupHandler).ServeHTTP()
      /home/shippable/.gvm/gos/go1.4/src/net/http/httptest/server.go:200 +0xf7
  net/http.serverHandler.ServeHTTP()
      /home/shippable/.gvm/gos/go1.4/src/net/http/server.go:1703 +0x1f6
  net/http.(*conn).serve()
      /home/shippable/.gvm/gos/go1.4/src/net/http/server.go:1204 +0x1087
==================
==================
WARNING: DATA RACE
Read by goroutine 149:
  github.com/docker/spdystream.(*Stream).Reset()
      /home/shippable/workspace/src/github.com/GoogleCloudPlatform/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/stream.go:170 +0x4d
  github.com/docker/spdystream.(*idleAwareFramer).monitor()
      /home/shippable/workspace/src/github.com/GoogleCloudPlatform/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:76 +0x35a

Previous write by goroutine 150:
  github.com/docker/spdystream.(*Connection).addStreamFrame()
      /home/shippable/workspace/src/github.com/GoogleCloudPlatform/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:363 +0x222
  github.com/docker/spdystream.(*Connection).Serve()
      /home/shippable/workspace/src/github.com/GoogleCloudPlatform/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:261 +0xd06

Goroutine 149 (running) created at:
  github.com/docker/spdystream.NewConnection()
      /home/shippable/workspace/src/github.com/GoogleCloudPlatform/kubernetes/Godeps/_workspace/src/github.com/docker/spdystream/connection.go:184 +0x757
  github.com/GoogleCloudPlatform/kubernetes/pkg/util/httpstream/spdy.NewServerConnection()
      /home/shippable/workspace/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/util/httpstream/spdy/connection.go:54 +0x6e
  github.com/GoogleCloudPlatform/kubernetes/pkg/util/httpstream/spdy.responseUpgrader.UpgradeResponse()
      /home/shippable/workspace/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/util/httpstream/spdy/upgrade.go:71 +0x530
  github.com/GoogleCloudPlatform/kubernetes/pkg/util/httpstream/spdy.(*responseUpgrader).UpgradeResponse()
      <autogenerated>:1 +0xf5
  github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet.(*Server).handleExec()
      github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/_test/_obj_test/server.go:498 +0x943
  github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet.*Server.(github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet.handleExec)·fm()
      github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/_test/_obj_test/server.go:95 +0x52
  net/http.HandlerFunc.ServeHTTP()
      /home/shippable/.gvm/gos/go1.4/src/net/http/server.go:1265 +0x4e
  net/http.(*ServeMux).ServeHTTP()
      /home/shippable/.gvm/gos/go1.4/src/net/http/server.go:1541 +0x20c
  github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet.(*Server).ServeHTTP()
      github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/_test/_obj_test/server.go:733 +0x1f0
  net/http/httptest.(*waitGroupHandler).ServeHTTP()
      /home/shippable/.gvm/gos/go1.4/src/net/http/httptest/server.go:200 +0xf7
  net/http.serverHandler.ServeHTTP()
      /home/shippable/.gvm/gos/go1.4/src/net/http/server.go:1703 +0x1f6
  net/http.(*conn).serve()
      /home/shippable/.gvm/gos/go1.4/src/net/http/server.go:1204 +0x1087

Goroutine 150 (running) created at:
  github.com/GoogleCloudPlatform/kubernetes/pkg/util/httpstream/spdy.newConnection()
      /home/shippable/workspace/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/util/httpstream/spdy/connection.go:68 +0x17a
  github.com/GoogleCloudPlatform/kubernetes/pkg/util/httpstream/spdy.NewServerConnection()
      /home/shippable/workspace/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/util/httpstream/spdy/connection.go:60 +0x117
  github.com/GoogleCloudPlatform/kubernetes/pkg/util/httpstream/spdy.responseUpgrader.UpgradeResponse()
      /home/shippable/workspace/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/util/httpstream/spdy/upgrade.go:71 +0x530
  github.com/GoogleCloudPlatform/kubernetes/pkg/util/httpstream/spdy.(*responseUpgrader).UpgradeResponse()
      <autogenerated>:1 +0xf5
  github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet.(*Server).handleExec()
      github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/_test/_obj_test/server.go:498 +0x943
  github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet.*Server.(github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet.handleExec)·fm()
      github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/_test/_obj_test/server.go:95 +0x52
  net/http.HandlerFunc.ServeHTTP()
      /home/shippable/.gvm/gos/go1.4/src/net/http/server.go:1265 +0x4e
  net/http.(*ServeMux).ServeHTTP()
      /home/shippable/.gvm/gos/go1.4/src/net/http/server.go:1541 +0x20c
  github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet.(*Server).ServeHTTP()
      github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/_test/_obj_test/server.go:733 +0x1f0
  net/http/httptest.(*waitGroupHandler).ServeHTTP()
      /home/shippable/.gvm/gos/go1.4/src/net/http/httptest/server.go:200 +0xf7
  net/http.serverHandler.ServeHTTP()
      /home/shippable/.gvm/gos/go1.4/src/net/http/server.go:1703 +0x1f6
  net/http.(*conn).serve()
      /home/shippable/.gvm/gos/go1.4/src/net/http/server.go:1204 +0x1087
==================
--- FAIL: TestServeExecInContainerIdleTimeout (0.14s)
	server_test.go:662: error creating input stream: Stream reset
E0227 19:55:03.766734   30657 server.go:552] Timed out waiting for client to create streams
W0227 19:55:04.338883   30657 connection.go:126] Stream rejected: Unable to parse '' as a port: strconv.ParseUint: parsing "": invalid syntax
W0227 19:55:04.520106   30657 connection.go:126] Stream rejected: Unable to parse 'abc' as a port: strconv.ParseUint: parsing "abc": invalid syntax
W0227 19:55:04.632877   30657 connection.go:126] Stream rejected: Unable to parse '-1' as a port: strconv.ParseUint: parsing "-1": invalid syntax
W0227 19:55:04.902832   30657 connection.go:126] Stream rejected: Unable to parse '65536' as a port: strconv.ParseUint: parsing "65536": value out of range
W0227 19:55:05.050841   30657 connection.go:126] Stream rejected: Port '0' must be greater than 0
FAIL
```

--- a/Godeps/Godeps.json
+++ b/Godeps/Godeps.json
@@ -114,7 +114,7 @@
 		},
 		{
 			"ImportPath": "github.com/docker/spdystream",
-			"Rev": "29e1da2890f60336f98d0b3bf28b05070aa2ee4d"
+			"Rev": "e9bf9912b85eec0ed6aaf317808a0eab25e3ca43"
 		},
 		{
 			"ImportPath": "github.com/elazarl/go-bindata-assetfs",

--- a/Godeps/_workspace/src/github.com/docker/spdystream/connection.go
+++ b/Godeps/_workspace/src/github.com/docker/spdystream/connection.go
@@ -72,11 +72,17 @@ Loop:
 				timer.Reset(i.timeout)
 			}
 		case <-expired:
-			for _, stream := range i.conn.streams {
-				stream.Reset()
-			}
-			i.conn.Close()
-			break Loop
+			i.conn.streamCond.L.Lock()
+			streams := i.conn.streams
+			i.conn.streams = make(map[spdy.StreamId]*Stream)
+			i.conn.streamCond.Broadcast()
+			i.conn.streamCond.L.Unlock()
+			go func() {
+				for _, stream := range streams {
+					stream.resetStream()
+				}
+				i.conn.Close()
+			}()
 		case <-i.conn.closeChan:
 			if timer != nil {
 				timer.Stop()

--- a/Godeps/_workspace/src/github.com/docker/spdystream/spdy_test.go
+++ b/Godeps/_workspace/src/github.com/docker/spdystream/spdy_test.go
@@ -396,6 +396,49 @@ func TestCloseNotification(t *testing.T) {
 	}
 }
 
+func TestIdleShutdownRace(t *testing.T) {
+	var wg sync.WaitGroup
+	server, listen, serverErr := runServer(&wg)
+	if serverErr != nil {
+		t.Fatalf("Error initializing server: %s", serverErr)
+	}
+
+	conn, dialErr := net.Dial("tcp", listen)
+	if dialErr != nil {
+		t.Fatalf("Error dialing server: %s", dialErr)
+	}
+
+	spdyConn, spdyErr := NewConnection(conn, false)
+	if spdyErr != nil {
+		t.Fatalf("Error creating spdy connection: %s", spdyErr)
+	}
+	go spdyConn.Serve(NoOpStreamHandler)
+
+	authenticated = true
+	stream, err := spdyConn.CreateStream(http.Header{}, nil, false)
+	if err != nil {
+		t.Fatalf("Error creating stream: %v", err)
+	}
+
+	spdyConn.SetIdleTimeout(5 * time.Millisecond)
+	go func() {
+		time.Sleep(5 * time.Millisecond)
+		stream.Reset()
+	}()
+
+	select {
+	case <-spdyConn.CloseChan():
+	case <-time.After(20 * time.Millisecond):
+		t.Fatal("Timed out waiting for idle connection closure")
+	}
+
+	closeErr := server.Close()
+	if closeErr != nil {
+		t.Fatalf("Error shutting down server: %s", closeErr)
+	}
+	wg.Wait()
+}
+
 func TestIdleNoTimeoutSet(t *testing.T) {
 	var wg sync.WaitGroup
 	server, listen, serverErr := runServer(&wg)
@@ -550,6 +593,44 @@ Loop:
 			t.Fatal("Timed out waiting for idle connection closure")
 		}
 	}
+
+	closeErr := server.Close()
+	if closeErr != nil {
+		t.Fatalf("Error shutting down server: %s", closeErr)
+	}
+	wg.Wait()
+}
+
+func TestIdleRace(t *testing.T) {
+	var wg sync.WaitGroup
+	server, listen, serverErr := runServer(&wg)
+	if serverErr != nil {
+		t.Fatalf("Error initializing server: %s", serverErr)
+	}
+
+	conn, dialErr := net.Dial("tcp", listen)
+	if dialErr != nil {
+		t.Fatalf("Error dialing server: %s", dialErr)
+	}
+
+	spdyConn, spdyErr := NewConnection(conn, false)
+	if spdyErr != nil {
+		t.Fatalf("Error creating spdy connection: %s", spdyErr)
+	}
+	go spdyConn.Serve(NoOpStreamHandler)
+
+	spdyConn.SetIdleTimeout(10 * time.Millisecond)
+
+	authenticated = true
+
+	for i := 0; i < 10; i++ {
+		_, err := spdyConn.CreateStream(http.Header{}, nil, false)
+		if err != nil {
+			t.Fatalf("Error creating stream: %v", err)
+		}
+	}
+
+	<-spdyConn.CloseChan()
 
 	closeErr := server.Close()
 	if closeErr != nil {

--- a/Godeps/_workspace/src/github.com/docker/spdystream/stream.go
+++ b/Godeps/_workspace/src/github.com/docker/spdystream/stream.go
@@ -168,7 +168,10 @@ func (s *Stream) Close() error {
 // Reset sends a reset frame, putting the stream into the fully closed state.
 func (s *Stream) Reset() error {
 	s.conn.removeStream(s)
+	return s.resetStream()
+}
 
+func (s *Stream) resetStream() error {
 	s.finishLock.Lock()
 	if s.finished {
 		s.finishLock.Unlock()

